### DISCUSSION TITLE
Extending "Find All References" and "Rename" functionality

### DISCUSSION THREAD

**Discussion by VolodymyrBS** [dotnet/roslyn#53519] [Q&A]
Hello everyone!

I want to write Visual Studio extension to search methods and classes in Unity prefab and scenes.
But to do it I need a some way extend Find all reference result with my own search result. 
Also it will be cool if I can get some kind of callback about symbol rename to update class/method name in Unity files.

So my question if it possible to extend reference search with custom search result?
and if it possible to get callback about symbol rename?

Sorry if it's wrong repo for such question.

Thanks in advance.

**Comment by ryzngard** [comment]
We don't have hooks for rename directly in Visual Studio from Roslyn, but there's a standard way that components get notified. We trigger [IVSRefactorNotify](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivsrefactornotify?view=visualstudiosdk-2019) operations for when we rename symbols. You should be able to use that as the event to update unity files. 

What are you thinking of specifically to add to Find All References?

**Reply by VolodymyrBS** [reply]
Thanks for your reply. That interface looks like the right solution for me!

Talking about "Find All References" I just want to display some additional references, not from C# code. In my case I want to display where Unity has reference to class/method.
Yes, I could do it by using some other combination of keys, but using standard "Find All References" looks better IMO.

**Reply by VolodymyrBS** [reply]
BTW looks like Visual Studio for Unity already done something similar in a later version for types only.  it uses Alt+2. and implemented as CodeLens provider
![image](https://user-images.githubusercontent.com/20375870/122581472-c0cc0a00-d05f-11eb-9692-6fab375370f5.png)

**Reply by ryzngard** [reply]
Yea, it would be hard to extend FAR since it isn't designed that way, but a codelens provider that does this is beautiful. Glad Unity has done that :)
